src/main/webapp/app/admin/report-management/report-management.controller.js

```diff
@@ -25,72 +25,98 @@
         vm.reverse = pagingParams.ascending;
         vm.itemsPerPage = 5;
         vm.transition = transition;
-        
-        
-        vm.parseCron=parseCron;
-        vm.executeNow=executeNow;
-        vm.goToBuildPage=goToBuildPage;
-        vm.deleteReport=deleteReport;
-        vm.updateReport=updateReport;
-        vm.searchReports=searchReports;
-        vm.openCalendar=openCalendar;
-        vm.datePickerOpenStatus={};
+
+
+        vm.parseCron = parseCron;
+        vm.executeNow = executeNow;
+        vm.goToBuildPage = goToBuildPage;
+        vm.deleteReport = deleteReport;
+        vm.updateReport = updateReport;
+        vm.searchReports = searchReports;
+        vm.saveSMTPSetting = saveSMTPSetting;
+
+        vm.deleteWebhook = deleteWebhook;
+        vm.openTeamConfigDialog = openTeamConfigDialog;
+        vm.reportManagementTabClick = reportManagementTabClick;
+        vm.getWebhookList = getWebhookList;
+        vm.openCalendar = openCalendar;
+        vm.datePickerOpenStatus = {};
         vm.datePickerOpenStatus.fromDate = false;
         vm.datePickerOpenStatus.toDate = false;
-        vm.dateFormat='yyyy-MM-dd';
-        vm.user=null;
+        vm.webhookList = [];
+        vm.SMPTSetting = {};
+        vm.dateFormat = 'yyyy-MM-dd';
+        vm.user = null;
+        vm.emailChannelConfig = [];
+        vm.teamChannelConfig = [];
+        vm.connection = {};
 
         activate();
         ///////////////////////////////////////
 
         function activate() {
-            getAccount(); 
-            getScheduledReports(vm.account.login,"","",""); 
-            var cronstrue = window.cronstrue; 
-             
+            reportManagementTabClick('report');
+            var cronstrue = window.cronstrue;
+        }
+
+        function reportManagementTabClick(tabName) {
+            if (tabName == "report") {
```

Comment:
please use command design patter instead of muliple if else...u can refer recentlyAccessedController

Reply:
in recentlyAccessedController every tab has same modal 
but here every tab has diffrent data like email and jira have only one record but team have mutiple data 
and every channnel has differnt property

Reply:
u can do this as below

         var config=   'email': {
                getData: function () {
                  
                }
            },
            'team': {
                getData: function () {
                    
                }
}
config[id].getData()